static/js/compose_validate.js

```diff
@@ -191,27 +194,35 @@ export function warn_if_topic_resolved(topic_changed) {
     const stream_name = compose_state.stream_name();
     const message_content = compose_state.message_content();
     const sub = stream_data.get_sub(stream_name);
-    const $resolved_notice_area = $("#compose_resolved_topic");
+    const $compose_banner_area = $("#compose_banners");
 
     if (sub && message_content !== "" && resolved_topic.is_resolved(topic_name)) {
-        if ($resolved_notice_area.html()) {
+        if ($("#compose_resolved_topic").length) {
             // Error is already displayed; no action required.
             return;
         }
 
+        const button_text = settings_data.user_can_move_messages_between_streams()
+            ? $t({defaultMessage: "Unresolve topic"})
+            : null;
+
         const context = {
+            banner_type: WARNING,
             stream_id: sub.stream_id,
             topic_name,
-            can_move_topic: settings_data.user_can_move_messages_between_streams(),
+            banner_text: $t({
+                defaultMessage:
+                    "You are sending a message to a resolved topic. You can send as-is or unresolve the topic first.",
+            }),
+            button_text,
+            css_id: "compose_resolved_topic",
         };
 
-        const new_row = render_compose_resolved_topic(context);
```

Comment:
I think you missed deleting the HTML template for this, which is helpful to look at, as it reveals where the strings come from.